plugin.go

```diff
@@ -263,7 +263,12 @@ func (s *Plugin) Middleware(next http.Handler) http.Handler {
 	var server FileServer = server
 
 	if s.cfg.Immutable {
-		server = createImmutableServer(s)
+		immutableServer, err := createImmutableServer(s)
+		if err != nil {
+			panic(err)
```

Comment:
Panic in Go is not the same as exception in PHP, for example, and should be avoided as much as possible. All data should be initialized (since you know the list of files to serve on the Init stage) on the Init stage. Middleware, on error should delete the buggy file from the list of the served files (or try to re-fetch).

Reply:
Ah I see. This way should be ok then?